base/XGamepad.ts

```diff
@@ -0,0 +1,503 @@
+    name?: string;
+    /** For multi-buttons, raw button combinaison */
+    as?: string;
+    /** Button code of the controller button that triggers the game button */
+    button_code?: number;
+    /** Button codes of the controller buttons that trigger the game button, should match {@link KeyMap#game_buttons} */
+    button_codes?: number[];
+    /** Keycode of the keyboard key that triggers the game button */
+    key_code?: number;
+    /** Modifiers needed along the {@link KeyMap#key_code} */
+    key_modifiers?: {alt?: boolean; ctrl?: boolean; shift?: boolean};
+};
+
+export class GamepadButton {
+    /** Emulated gamepad controller */
+    gamepad: Gamepad;
+    /** Game button that will trigger */
+    button: Button | CButton;
+    /** Debug purpose */
+    name: string = undefined;
+    /** Whether the gamepad button is held down */
+    is_down: boolean = false;
+    /** Signal to trigger when that button is pressed */
+    on_down = new Phaser.Signal();
+    /** Signal to trigger when that button is release */
+    on_up = new Phaser.Signal();
+
+    constructor(gamepad: Gamepad, button: Button | CButton) {
+        this.gamepad = gamepad;
+        this.button = button;
+        this.name = Button[this.button] ?? CButton[this.button];
+    }
+    /** Whether the gamepad button is up */
+    get is_up() {
+        return !this.is_down;
+    }
+    set is_up(v: boolean) {
+        this.is_down = !v;
+    }
+}
+
+/**
+ * An emulated gamepad.
+ */
+export class Gamepad {
+    /** Default gamepad buttons configuration */
+    static gamepad_mapping: KeyMap[];
+    /** Custom gamepad buttons configuration */
+    static gamepad_stick_mapping: KeyMap[];
+    /** Custom gamepad buttons configuration */
+    static keyboard_mapping: KeyMap[];
+    /**
+     * Loads the gamepad and keyboard configuration.
+     * @param {GoldenSun} data - Master game data object
+     * @param {string} [input_type=default_inputs] - Keyboard key mapping configuration name
+     */
+    static initialize(data: GoldenSun, input_type: string = "default_inputs") {
+        // Load default gamepad buttons configuration
+        const gamepad_mapping = Object.entries(data.dbs.init_db.gamepad_rinputs as {[code: string]: string}).map(
+            ([button_code, game_button]): KeyMap => ({
+                name: game_button,
+                game_button: Button[game_button] ?? CButton[game_button],
+                button_code: Phaser.Gamepad[button_code],
+            })
+        );
+        // Load custom gamepad buttons configuration
+        const gamepad_custom_mapping = Object.entries(data.dbs.init_db.gamepad_inputs as {[code: string]: string}).map(
+            ([game_button, button_code]): KeyMap => {
+                const matches = button_code.match(/^(?:(?:(\w+) *\+ *)?(\w+) *\+ *)?(\w+)$/);
+                if (!matches) console.error("Input not recognized " + button_code);
+                const get_button_code = (code: string): number =>
```

Comment:
Considering that this line break is due to the formatter, there's no problem about using curly braces in your arrow function in order to reduce code width and avoid it, it also improves readability. Readability always comes first.